src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -644,43 +644,58 @@ void GenerateFieldAccessor(const FieldDescriptor* field, const Options& options,
   // Generate getter.
   GenerateFieldDocComment(printer, field, options, kFieldGetter);
 
+  // deprecation
+  std::string deprecation_std_string = (field->options().deprecated()) ? "@trigger_error('" +
+      field->name() +
+      " is deprecated and will be removed in the next major release', E_USER_DEPRECATED);" : "";
+  const char* deprecation_string = deprecation_std_string.c_str();
+
   if (oneof != NULL) {
     printer->Print(
         "public function get^camel_name^()\n"
         "{\n"
+        "    ^deprecation_trigger^\n"
```

Comment:
Please collapse the newlines so that a non-deprecated field has the same generated code as today.

You can run `generated_descriptor_proto.sh` in the base directory to regenerate the PHP files with the new compiler.

Reply:
I get what you mean, but I do not know how. I ended up with this solution to have a properly formatted output file, whether or not a field is deprecated.

Can you please give me a hint?

Reply:
Maybe try:

```c++
    std::string deprecation_warning;
    if (field->options().deprecated()) {
        // Note trailing newline and indenting spaces.
        deprecation_warning = "@trigger_error('" +
             field->name() + " is deprecated', E_USER_DEPRECATED);\n    ";
    }
    printer->Print(
        "public function get^camel_name^()\n"
        "{\n"
        "    ^deprecation_warning^return $this->readOneof(^number^);\n"
        "}\n\n",
        "camel_name", UnderscoresToCamelCase(field->name(), true),
        "number", IntToString(field->number()),
        "deprecation_warning", deprecation_warning);
```

Reply:
Thanks, this helped.